src/app/submissions/submissions.service.ts

```diff
@@ -463,17 +474,10 @@ export class SubmissionsService {
           const planetsWithName = attachNamesToPlanets(planets);
           const submissionsWithPlanetName = filteredSubmissions.map(submission => ({
             ...submission,
-            planetName: codeToPlanetName(submission.source, this.stateService.configuration, planetsWithName)
+            planetName: codeToPlanetName(submission.source, this.stateService.configuration, planetsWithName),
+            teamInfo: submission.team ? { name: submission.team.name, type: submission.team.type } : null
           }));
-          return forkJoin(
-            submissionsWithPlanetName.map(submission => {
-              if (submission.team) {
-                return this.teamsService.getTeamName(submission.team).pipe(map(teamInfo => ({ ...submission, teamInfo })));
-              }
-              return of(submission);
-            })
-          ).pipe(map((updatedSubmissions: any[]): [ any[], number, string[] ] => [ updatedSubmissions, time, questionTexts ])
-          );
+          return of<[any[], number, string[]]>([submissionsWithPlanetName, time, questionTexts]);
```

Comment:
Same, since we're no longer pulling in team info here via another observable we can just use a map instead of switchMap and return the array.